app/main.py

```diff
@@ -113,8 +116,55 @@ async def upload_image(file: UploadFile = File(...)):
         ])
 
 
-@app.get("/track_growth")
-async def track_growth():
-    raise HTTPException(
-        status_code=status.HTTP_501_NOT_IMPLEMENTED
-    )
+@app.get("/track_growth/")
+async def track_growth(center: Tuple[float, float] = None, scan_range: float = 360, day_range: int = 1000):
+
+    if center is None:
+        center = (0, 0)
+
+    SQL = f"""
+    SELECT 
+        plant_id, 
+        ARRAY_AGG(
+            ARRAY[
+                TO_CHAR(DATE_TRUNC('day', date), 'YYYY-MM-DD'), 
+                latitude::text,
+                longitude::text,
+                count::text
+            ] ORDER BY date
+        ) AS date_counts
+    FROM (
+        SELECT 
+            t1.plant_id, 
+            t2.date, 
+            t2.latitude,
+            t2.longitude,
+            COUNT(*) AS count
+        FROM 
+            image_processing.processed_entry t1
+        JOIN 
+            image_processing.raw_entry t2 ON t1.image_uri = t2.image_uri
+        WHERE
+            (({center[0]} + t2.latitude)^2 + ({center[1]} + t2.longitude)^2) <= {scan_range}^2
+            AND t2.date BETWEEN CURRENT_DATE - {day_range} AND CURRENT_DATE 
```

Comment:
What is the extra `AND CURRENT_DATE` doing?

Reply:
It's checking for all entries between (current date - day range) and the current date